Create kong-service-manager.yaml

ValberJunior opened the pull request:
## Hey, I just made a Pull Request!

### Kong Service Manager Plugin

The Kong Service Manager plugin provides information about the kong service, a list of the routes it has and also offers the possibility of manipulating plugins without leaving the backstage.

**About Page**
![image](https://github.com/backstage/backstage/assets/84424883/ea8f84f3-fc85-4128-ac1c-2d493f37c42e)

**All Routes**
On this screen, we list all the routes that the service has:

![image](https://github.com/backstage/backstage/assets/84424883/2cb08227-2f72-42be-a526-dd95bf01eb86)

**Plugins List**
Finally, we have the list of plugins:

List of all plugins eligible for your Kong instance; ✅
Installing / Removing and Editing a plugin at your Service; ✅
Plugin search field; ✅
Filtering by the plugins installed in the service; ✅

![image](https://github.com/backstage/backstage/assets/84424883/02a687e6-7abf-4a51-8c44-f798b5d27c1d)

**Configuring a plugin in Service**

For the example, we'll use the rate limiting plugin:
![image](https://github.com/backstage/backstage/assets/84424883/387d69b8-61b8-40e2-9823-7c9112247b7a)

When you install it, it will appear on the Associated Plugins tab:
![image](https://github.com/backstage/backstage/assets/84424883/59e7466c-98fa-4804-a4f2-f1e5079702df)

From then on, the plugin will already be configured in your service:
![image](https://github.com/backstage/backstage/assets/84424883/7f642412-3797-49ce-8193-58c4433509f8)


#### :heavy_check_mark: Checklist

<!--- Please include the following in your Pull Request when applicable: -->

- [ :heavy_check_mark:  ] Added or updated documentation
- [ :heavy_check_mark: ] Screenshots attached (for UI changes)
- [ :heavy_check_mark:] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

vinzscam reviewed (changes requested):
Thank you! Two things:
- Is it possible to make more visible where the [source code](https://github.com/veecode-platform/platform-backstage-plugins/tree/master/plugins/kong-service-manager) is? This is important so that people know where to report issues
- I think something is odd in the configuration of the plugin, in section "2- Additional settings": https://github.com/veecode-platform/platform-backstage-plugins/tree/master/plugins/kong-service-manager#configuration-%EF%B8%8F
ideally no information regarding the proxy configuration should be exposed to the frontend

ValberJunior commented:
@vinzscam 
Hi, thanks for the feedback.
Regarding the first question, I fixed an update and added a link to our github community in the documentation.
Regarding the proxy, we only needed to retrieve the target and workspace, I even edited `config.d.ts` to access only this information:

```ts
export interface Config {
  /**
  * @visibility frontend
  */
  proxy?: {
    /** @visibility frontend */
    endpoints?: {
      /** @visibility frontend */
      [key: string]:
        | string
        | {
             /** @visibility frontend */
              target: string;
             /** @visibility frontend */
              workspace?: string;
          };
    };
  } 
  }
  /**
``` 
But I understand that it's not ideal to expose the proxy on the frontend, so I'll provide a fix to reference the workspace via annotation.

ValberJunior reviewed (commented):
As reported,

- I have changed the documentation with a banner containing the community link on github.
- I removed the exposure from the proxy, adopting the annotations approach

vinzscam commented:
> As reported,
> 
> * I have changed the documentation with a banner containing the community link on github.
> * I removed the exposure from the proxy, adopting the annotations approach

Thank you, it looks great! I guess this file https://github.com/veecode-platform/platform-backstage-plugins/blob/master/plugins/kong-service-manager/config.d.ts is not needed anymore right?

ValberJunior commented:
@vinzscam  correct, this file is no longer used, we removed it from the repository :heavy_check_mark:

Rugvip reviewed (approved):
Thank you!

As a side note I'd encourage you to replace the proxy configuration with a backend plugin that is responsible for routing requests to the correct instance. That'd let you limit the use of credentials better.
